stack/framework/hal/platforms/PUSH7/platf_button.c

```diff
@@ -62,11 +63,26 @@ __LINK_C void __ubutton_init()
         assert(err == SUCCESS);
     }
     // gather all button states
-    for (uint8_t i = 0; i < PLATFORM_NUM_BUTTONS; i++) 
-    {
-        booted_button_state += hw_gpio_get_in(buttons[i].button_id) << i;
-    }
+        for (uint8_t i = 0; i < PLATFORM_NUM_BUTTONS; i++) 
+        {
+            bool button_state = hw_gpio_get_in(buttons[i].button_id);
+            booted_button_state += button_state << i;
+            buttons[i].last_known_state = button_state;
+        }
+    
     sched_register_task(&button_task);
+    sched_register_task(&validate_state);
+    sched_post_task(&validate_state);
+}
+
+static void validate_state()
```

Comment:
why is this here?

Reply:
When a button is pressed and released during reset, the button will be read out as high right after boot. When reading it out again afterwards, it will return the actual state of the GPIO.